lib/graphql/input_object_type.rb

```diff
@@ -51,6 +51,18 @@ def kind
     def validate_non_null_input(input, warden)
       result = GraphQL::Query::InputValidationResult.new
 
+      if input.is_a? Array
```

Comment:
Haha no worries! I think the normalize solution is best. Calling `.to_h` on an array will only work if it's an array of tuples:
```
[1].to_h
=> TypeError: wrong element type Fixnum at 0 (expected array)

[[1,2,3,4]].to_h
=> ArgumentError: wrong array length at 0 (expected 2, was 4)
```
So the entire solution would be something like
```
val_h = value.is_a?(Hash) ? value : (value.to_h rescue nil)
return if val_h.nil?
```